[Dependency Cache] Fix Dependency Cache

ParaskP7 opened the pull request:
Related: #135

### Description

As part of #135 and this last 1dda7bd51d20fbb82d5435dc2cbee3b08f5b5153 commit I incorrectly removed the `DEP_CACHE_FOLDER_NAME` variable. Without this variable set, placing Gradle dependency cache is not working as expected due to the fact that this `DEP_CACHE_FOLDER_NAME` variable is now unbound. And this makes this whole dependency cache solution not working at all.

---

### Testing Instructions

Base on this [WCAndroid##13303](https://github.com/woocommerce/woocommerce-android/pull/13303) PR and this [test commit](https://github.com/woocommerce/woocommerce-android/pull/13303/commits/49d771a9cf4f11e7efdf7dd0641e2b529bf39f07), looking at its corresponding [build](https://buildkite.com/automattic/woocommerce-android/builds/26598):
1. Notice that during the [cache restoration](https://buildkite.com/automattic/woocommerce-android/builds/26598#01946985-7315-47ec-bea7-01814e8a7755/189-190) process on a specific job (ie. `Mobile App`) the [cache is being placed](https://buildkite.com/automattic/woocommerce-android/builds/26598#01946985-7315-47ec-bea7-01814e8a7755/189-200) as expected.
2. Looking at the [network activity](https://scans.gradle.com/s/aqheikr2epk5u/performance/network-activity) of a specific job (ie. `Mobile App`), make sure that the network activity is minimal and close to zero (ie. `Wall clock time spent on network requests | 0s`.

---

- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

ParaskP7 committed:
Gradle: Fix dependency cache

As part of #135 and this last 1dda7bd51d20fbb82d5435dc2cbee3b08f5b5153
commit I incorrectly removed the 'DEP_CACHE_FOLDER_NAME' variable.
Without this variable set, placing Gradle dependency cache is not
working as expected due to the fact that this 'DEP_CACHE_FOLDER_NAME'
variable is now unbound. And this makes this whole dependency cache
solution not working at all.

dangermattic commented:
<!--
  0 Errors
  1 Warning: Please add an entry in the <co...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please add an entry in the <code>CHANGELOG.md</code> file to describe the changes made by this PR</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 committed:
Release: Update `README` and `CHANGELOG` with 3.9.0 release info (#135)

FYI: Lint is now failing otherwise: https://buildkite.com/automattic/
ci-toolkit-buildkite-plugin/builds/
894#01946971-8ca9-4e6b-8024-da9e25f85121

PS: I wasn't aware about this process and thus missed updating both, the
'README' and 'CHANGELOG' with 3.9.0 release info after merging #135 by
creating a separate 'release/3.9.0' branch and PR for release purposes.
Instead, I just went and drafted a new release directly from GitHub and
the project's releases' page.

Example Release Process Related PR: https://github.com/Automattic/
a8c-ci-toolkit-buildkite-plugin/pull/137

iangmaia commented on README.md:
```diff
@@ -26,7 +26,7 @@ steps:
 
     plugins:
-      - automattic/a8c-ci-toolkit#3.8.0:
+      - automattic/a8c-ci-toolkit#3.9.0:
```
I see a `3.9.0` release already in the repo, perhaps it was created too early? Indeed it's best to make a new release following https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin?tab=readme-ov-file#releasing. Perhaps you can delete that one with its tag as well? 🤔

wzieba replied:
Yeah, I feel like it should be `3.9.1`, right?

wzieba replied:
Or maybe even this should be unchanged in this PR, and only bumped in the release PR?

ParaskP7 replied:
👋 @iangmaia and @wzieba yea, I wasn't aware about the release process and thought that we are just creating a tag. Maybe let's keep this [3.9.0](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/releases/tag/3.9.0) release as is and I'll create a new `3.9.1` release right now, following the release process, after merging this PR. Wdyt? 🤔

ParaskP7 replied:
> Or maybe even this should be unchanged in this PR, and only bumped in the release PR?

Lint fails otherwise, and actually @AliSoftware did the same on this #139 that just got merged (plus [this](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/139#issuecomment-2592304957)). 😊

iangmaia reviewed (approved):
👍

ParaskP7 committed:
Merge branch 'trunk' of github.com:Automattic/a8c-ci-toolkit-buildkite-plugin into gradle/fix-dependency-cache

ParaskP7 committed:
Release: Add 'CHANGELOG' entry
